File: app/src/main/java/com/firebaseapp/sowbreira_26fe1/fl_mane/GridActivity.java
package com.firebaseapp.sowbreira_26fe1.fl_mane;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.GridView;

public class GridActivity extends AppCompatActivity {

    private int itemPosition = -1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_grid);
        final GridView gridview = (GridView) findViewById(R.id.gridview);
        gridview.setAdapter(new ImageAdapter(this));

        gridview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            public void onItemClick(AdapterView<?> parent, View v,
                                    int position, long id) {
                gridview.setItemChecked(position, true);
                itemPosition = position;
            }
        });

        findViewById(R.id.ok_foto).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if(itemPosition<0){
                    return;
                }
                SharedPreferences settings = getSharedPreferences(LoginActivity.PREFS_NAME, 0);
                SharedPreferences.Editor editor = settings.edit();
                editor.putString("foto", "https://sowbreira-26fe1.firebaseapp.com/f1mane/profile/profile-"+itemPosition+".png");
                editor.commit();
                voltaPerfil();
            }
        });
        findViewById(R.id.cancel_foto).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                GridActivity.this.finish();
                voltaPerfil();
            }
        });

    }

    private void voltaPerfil() {
        Intent intent = new Intent(GridActivity.this, PerfilActivity.class);
        startActivity(intent);
        finish();
    }

}

File: app/src/main/java/com/firebaseapp/sowbreira_26fe1/fl_mane/GridActivity.kt
package com.firebaseapp.sowbreira_26fe1.fl_mane

import android.content.Intent
import android.os.Bundle

import androidx.appcompat.app.AppCompatActivity

import com.firebaseapp.sowbreira_26fe1.fl_mane.databinding.ActivityGridBinding

class GridActivity : AppCompatActivity() {

    private var itemPosition = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityGridBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.gridview.adapter = ImageAdapter(this)

        binding.gridview.setOnItemClickListener { _, _, position, _ ->
            binding.gridview.setItemChecked(position, true)
            itemPosition = position
        }

        binding.okFoto.setOnClickListener {
            if (itemPosition < 0) {
                return@setOnClickListener
            }
            val settings = getSharedPreferences(LoginActivity.PREFS_NAME, 0)
            settings.edit()
                .putString("foto", "https://sowbreira-26fe1.firebaseapp.com/f1mane/profile/profile-$itemPosition.png")
                .commit()
            voltaPerfil()
        }
        binding.cancelFoto.setOnClickListener {
            finish()
            voltaPerfil()
        }
    }

    private fun voltaPerfil() {
        val intent = Intent(this, PerfilActivity::class.java)
        startActivity(intent)
        finish()
    }
}
